### PR TITLE
[FrameworkBundle] Bump `symfony/http-kernel`

### LINKED ISSUE #58581: Symfony 7.2 crashes with Error: Uncaught TypeError: Symfony\Component\HttpFoundation\Request::setTrustedProxies(): Argument #2 ($trustedHeaderSet) must be of type int, array given
### Symfony version(s) affected

7.2.x-dev

### Description

```
Symfony\Component\ErrorHandler\Error\FatalError:
Error: Uncaught TypeError: Symfony\Component\HttpFoundation\Request::setTrustedProxies(): Argument #2 ($trustedHeaderSet) must be of type int, array given, called in /srv/api/vendor/symfony/http-kernel/Kernel.php on line 749 and defined in /srv/api/vendor/symfony/http-foundation/Request.php:528
Stack trace:
#0 /srv/api/vendor/symfony/http-kernel/Kernel.php(749): Symfony\Component\HttpFoundation\Request::setTrustedProxies(Array, Array)
#1 /srv/api/vendor/symfony/http-kernel/Kernel.php(170): Symfony\Component\HttpKernel\Kernel->preBoot()
#2 /srv/api/vendor/symfony/runtime/Runner/Symfony/HttpKernelRunner.php(35): Symfony\Component\HttpKernel\Kernel->handle(Object(Symfony\Component\HttpFoundation\Request))
#3 /srv/api/vendor/autoload_runtime.php(29): Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
#4 /srv/api/public/index.php(7): require_once('...')
#5 {main}
  thrown

  at /srv/api/vendor/symfony/http-foundation/Request.php:528                
```

most likely culprit: 60bd068d664864c03adf44e88896fa24a7c2e00a cc @nicolas-grekas 

### How to reproduce

framework.yaml
```yaml
framework:
 trusted_proxies: '%env(TRUSTED_PROXIES)%'
```
.env
```
TRUSTED_PROXIES=127.0.0.1,REMOTE_ADDR
```

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by alexandre-daubois**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Issues        | Fix #58581
| License       | MIT

Doing something conditionally (as stated in the issue) seems tricky because we deal with env var placeholders in `FrameworkExtension`. Another solution is to provide a bug fix to 7.1 where `HttpKernel/Kernel` would check if the `kernel.trusted_headers` parameter is null then fallback on a default value, but it doesn't feel quite right to me.

I think the best/cleanest solution is to simply bump the dep constraint in FWB's composer.json.

`symfony/serializer` must also be bumped to `^7.1`, tests fails otherwise: https://github.com/symfony/symfony/actions/runs/11384686629/job/31672929730?pr=58589#step:8:11669

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 1]
```diff
@@ -59,7 +59,7 @@
         "symfony/scheduler": "^6.4.4|^7.0.4",
         "symfony/security-bundle": "^6.4|^7.0",
         "symfony/semaphore": "^6.4|^7.0",
-        "symfony/serializer": "^6.4|^7.0",
+        "symfony/serializer": "^7.1",
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/composer.json:62]: Why is this change needed?

File: src/Symfony/Bundle/FrameworkBundle/composer.json [hunk 2]
```diff
@@ -95,7 +95,6 @@
         "symfony/property-access": "<6.4",
         "symfony/runtime": "<6.4.13|>=7.0,<7.1.6",
         "symfony/scheduler": "<6.4.4|>=7.0.0,<7.0.4",
-        "symfony/serializer": "<6.4",
```
Comment (xabbuh) [inline src/Symfony/Bundle/FrameworkBundle/composer.json:98]: `"symfony/serializer": "<7.1"` if we were to change the entry in the `require-dev` section

File: src/Symfony/Bundle/FrameworkBundle/composer.json (hunk 1 above)
Comment (alexandre-daubois) [reply src/Symfony/Bundle/FrameworkBundle/composer.json:62]: The test suite fails if serializer is < 7.1: https://github.com/symfony/symfony/actions/runs/11384686629/job/31672929730?pr=58589#step:8:11669

File: src/Symfony/Bundle/FrameworkBundle/composer.json (hunk 2 above)
Comment (alexandre-daubois) [reply src/Symfony/Bundle/FrameworkBundle/composer.json:98]: Ah yes, thanks!

File: src/Symfony/Bundle/FrameworkBundle/composer.json (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Bundle/FrameworkBundle/composer.json:62]: I wonder why this didn't fail before though. 🤔

Comment (stof) [reply src/Symfony/Bundle/FrameworkBundle/composer.json:62]: Maybe the behavior in ApiAttributeTests (switching the assertion based on the HttpKernel version being used) was not actually dependent only in the HttpKernel but also on the Serializer version, but no CI job was installing a recent HttpKernel with an old Serializer

**Review by xabbuh — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @alexandre-daubois.
